Sources/MapboxNavigation/NavigationViewControllerDelegate.swift

```diff
@@ -64,7 +64,18 @@ public protocol NavigationViewControllerDelegate: VisualInstructionDelegate {
      - returns: True to allow the navigation view controller to calculate a new route; false to keep tracking the current route.
      */
     func navigationViewController(_ navigationViewController: NavigationViewController, shouldRerouteFrom location: CLLocation) -> Bool
-    
+
+    /**
+     Called when the user arrives at a waypoint.
+
+     Return false to continue checking if reroute is needed. By default, the user will not be rerouted when arriving at a waypoint.
```

Comment:
The documentation for this method and `NavigationServiceDelegate.navigationService(_:shouldPreventReroutesWhenArrivingAt:)` should link to each other, so that developers who implement a custom UI don’t think they have to downgrade to NavigationViewController or vice versa to get this functionality.

Reply:
I will make a follow up update after discussion. I'm not sure why we need to link these two methods together.

Reply:
Both are public methods that are named similarly, which means a developer might accidentally find the documentation about one when they need the other one.